web/src/components/task/task-sidebar/task-sidebar.tsx

```diff
@@ -268,10 +269,19 @@ const TaskSidebar: FC<TaskSidebarProps> = ({ onRedirectAfterFinish, jobSettings,
         }
     }, [categories, latestLabelsId]);
 
-    const taxonomy = useLinkTaxonomyByCategoryAndJobId({
-        jobId,
-        categoryId: selectedAnnotation?.category!
-    });
+    let taxonomy;
+    if (
+        selectedCategory &&
+        Array.isArray(selectedCategory.data_attributes) &&
+        selectedCategory.data_attributes.some((category) => category.type === 'taxonomy')
+    ) {
+        taxonomy = useLinkTaxonomyByCategoryAndJobId({
+            jobId,
+            categoryId: selectedAnnotation?.category!
+        });
+    } else {
+        taxonomy = undefined;
```

Comment:
I suppose assigning the value "undefined" manually is not a good practice
https://learn.javascript.ru/types#:~:text=%E2%80%A6%D0%9D%D0%BE%20%D1%82%D0%B0%D0%BA%20%D0%B4%D0%B5%D0%BB%D0%B0%D1%82%D1%8C%20%D0%BD%D0%B5%20%D1%80%D0%B5%D0%BA%D0%BE%D0%BC%D0%B5%D0%BD%D0%B4%D1%83%D0%B5%D1%82%D1%81%D1%8F

Reply:
redefined to undefined to properly fire this condition(482/492 taxonomyId={(taxonomy?.data || [])[0]?.id}) when choosing a non-taxonomy option